tensornetwork/network_components.py

```diff
@@ -1171,6 +1171,7 @@ def _flatten_trace_edges(edges: List[Edge],
   edge2 = Edge(node1=node, axis1=len(perm_front) + 1, name="TraceBack")
   node.edges = node.edges[:len(perm_front)] + [edge1, edge2]
   new_edge = connect(edge1, edge2, new_edge_name)
+  node.axis_names = [str(n) for n in range(len(node.edges))]
```

Comment:
Hmm, I'm not sure this is what we actually want here. Edge names can sometimes be different from axis names. I think what we want is for all of the axes to keep their original name except for the ones that got flattened. And then for that name maybe go with a default name?

@amilsted @mganahl do you two have any opinions here?

Reply:
Yeah I see what you mean. We were ending up with nodes that had an `axis_names` list of a different size than the number of edges in the node which was breaking tests in my other PR. I thought about doing something similar to `flatten_edges`:

```
if axis_names:
      node.axis_names = [axis_names[n] for n in range(len(node.edges))]
    else:
      node.axis_names = [str(n) for n in range(len(node.edges))]
```

But that also seems to be not quite sufficient as it is only selecting the first n names instead of the correct n names. Also wasn't quite sure how we should determine the new axis name so I went with this solution to at least have the correct number of names.

Reply:
I think a better strategy would be to make a copy of the names in a new list and then remove all of the names in the indices of `edge.axis1` and `edge.axis2` for all of the given edges. You can see an example of the multi remove here. https://github.com/google/TensorNetwork/blob/e213de9ceae4a4003469cb65f78879bcb9e02514/tensornetwork/contractors/opt_einsum_paths/utils.py#L23